scripts/check-typescript-compatibility.js

```diff
@@ -0,0 +1,72 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const { printLog, runMain } = require('./lib/executionUtils')
+const { command } = require('./lib/command')
+
+const TEST_APP_DIR = path.join(__dirname, '..', 'test', 'app')
+
+runMain(() => {
+  printLog('Building project...')
+  command`yarn build`.run()
+
+  printLog('Setting up test environment...')
+  command`yarn`.withCurrentWorkingDirectory(TEST_APP_DIR).run()
+
+  const checks = [
+    {
+      title: 'TypeScript 3.8.2 compatibility',
+      version: '3.8.2',
+    },
+    {
+      title: 'TypeScript isolated modules compatibility',
+      compilerOptions: { isolatedModules: true },
+      version: '3.8.2',
+    },
+    {
+      title: 'TypeScript 4.1.6 compatibility',
+      version: '4.1.6',
+    },
+    {
+      title: 'TypeScript latest compatibility',
+      version: 'latest',
+    },
+    {
+      title: 'exactOptionalPropertyTypes compatibility',
+      version: 'latest', // Not available in 3.8.2
+      compilerOptions: { exactOptionalPropertyTypes: true },
+    },
+    {
+      title: 'ESNext compatibility',
+      version: 'latest',
+      compilerOptions: { lib: ['ESNext', 'DOM'] },
+    },
```

Comment:
All those tests were only one in the past. Keeping theme separated makes it easier to understand what fails.